src/coreclr/jit/emit.h

```diff
@@ -1793,15 +1795,27 @@ class emitter
 
         bool idIsEvexNdContextSet() const
         {
+            assert(!IsApxZuCompatibleInstruction(_idIns));
             return _idEvexNdContext != 0;
         }
 
+        bool idIsEvexZuContextSet() const
+        {
+            return (_idEvexZuContext != 0);
```

Comment:
nit: It looks like this is missing the `assert(IsApxZuCompatibleInstruction(_idIns))`